scripts/zen_workaround.py

```diff
@@ -43,7 +74,19 @@ def read_msr(cpu):
             val |= BIT
         msr = os.open('/dev/cpu/{}/msr'.format(cpu), os.O_WRONLY)
         os.lseek(msr, MSR, os.SEEK_SET)
-        os.write(msr, struct.pack('<q', val))
+        try:
+          os.write(msr, struct.pack('<q', val))
+        except PermissionError:
+            print("No permission to write to MSR. If Secure Boot is enabled, try disabling it, as it can deny user space writes to MSR by the Linux Kernel. Attempting to check if secure boot is enabled...")
+            check = is_secure_boot_enabled() 
+            if check == SecureBoot.enabled:
+                print("Secure boot returned 'enabled'.")
+            elif check == SecureBoot.disabled:
+                print("Secure boot returned 'disabled'")
+            else:
+                print("Secure boot status check failed.")
```

Comment:
I think we can simplify these outputs. If we know secure boot is enabled or disabled it doesn't make sense to print "If secure boot is enabled...".

So if it's enabled, print `Permission denied writing to MSR. Secure Boot is enabled, which causes this error. Try disabling Secure Boot?`.
If it's disabled, print `Permission denied writing to MSR. Secure Boot is disabled so that's not the problem.`
Otherwise print `Permission denied writing to MSR.`
How does that sound?

Reply:
Fixed it! Sounds great. Also this way the messages don't get repetitive and say basically more or less the same things twice (if we've established that secure boot is on, we've pretty much identified the reason).